### PR TITLE
migrate gradle build scripts from groovy to kotlin

### PR DESCRIPTION
**Pull request by uniqueid001**
.

### CODE REVIEW

**Comment by tuomas2** [comment]
- [ ] Related: https://github.com/AndBible/jsword/pull/6

File: app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    }
+    return stdout.toString().trim()
+        }) // use()
+}
+
+val npmUpgrade by tasks.registering(Exec::class) {
+    inputs.file("$jsDir/package.json")
+    outputs.file("$jsDir/node_modules/.bin/npm")
+    workingDir = file(jsDir)
+    // Workaround for F-droid, which has buggy npm version 5.8, that always fails when installing packages.
```
Comment (tuomas2) [inline app/build.gradle.kts:44]: Node is updated to version 14.16 in F-droid, and there should be more recent npm
 https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/net.bible.android.activity.yml

Comment (tuomas2) [reply app/build.gradle.kts:44]: ah, just noticed that this workaround is also in old version...

**Comment by tuomas2** [comment]
@uniqueid001 check the improvements made in the above mentioned commit ^

**Comment by tuomas2** [comment]
and thank you for doing this! This change has been in my mind. Good that we got it done now. I understand kotlin much better than groovy so it is easier to maintain.

File: app/build.gradle.kts (hunk 1 above)
Comment (uniqueid001) [reply app/build.gradle.kts:44]: : )
Yeah, I simply translated it.

**Comment by uniqueid001** [comment]
> @uniqueid001 check the improvements made in the above mentioned commit ^

I missed the nullable in the cast;  it compiled and I have the property defined so it was non-null at runtime too.
I see a number of style changes.  I am just learning kotlin myself, as I teach it to a homeschool class.  I am not accustomed to things like using an 'if' as an expression instead of a statement, or assigning a single expression to a function declaration.

**Pull request merged**